### ISSUE TITLE
Why drop_last in sft validation dataloader?

### ISSUE THREAD

**Issue by smahdavi4** [NVIDIA-NeMo/RL#958]
In the [SFT dataloader](https://github.com/NVIDIA-NeMo/RL/blob/38e9ef10be7424a03af669a89c001e8f25d3740c/nemo_rl/algorithms/sft.py#L148), the validation data loader has `drop_last=True`, is that a design choice or a bug? Since we want the validation loss be representative of all validation samples, would it make sense to set it to False, or the code logic would break?

**Comment by ashors1** [comment]
Hi, thanks for bringing this up. We should definitely support `drop_last=False`. I'll try to get a fix in for this ASAP.

**Comment by qiaochuz-nv** [comment]
## QA RCCA Analysis

RCCA Status: Done
RCCA Category: covered
Test Action: none

Fix PR: https://github.com/NVIDIA-NeMo/RL/pull/1029
Root Cause: fix PR includes upstream test coverage.
Trigger Config: Why drop_last in sft validation dataloader?

Coverage Assessment:
| Area | Result |
|---|---|
| Existing Test | Upstream fix PR test coverage |
| Coverage Status | COVERED |
| Gap | N/A |
| Corrective Action | No nmfw_tests addition required; fix PR includes upstream test coverage or equivalent regression coverage. |

Conclusion: RCCA classification completed for this closed NeMo-RL GitHub issue.
